src/test/java/seedu/address/logic/commands/ShowEventsCommandTests.java

```diff
@@ -0,0 +1,31 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for showfriends command.
+ */
+public class ShowEventsCommandTests {
```

Comment:
Can we include tests with different underlying `Model` states? 

The model could be permutations of:
1. Have people, but no events
2. Have no people, but somehow has events (obviously erroneous - `ShowEventsCommand` should notice this at execution and flag it, and fail.
3. Have people and events

Test cases should cover all and `ShowEventsCommand` should be flagging all erroneous cases, whether or not they could happen with other components.

Reply:
Thank you for suggesting these tests, I have added support for 1 and 3. However I feel that 2 should be stopped at adding events/ deleting friends, they are responsible for the upkeep of the relationship between friends and events.